jira/client.py

```diff
@@ -1232,7 +1232,17 @@ def group_members(self, group: str) -> OrderedDict:
 
         result = {}
         for user in r["users"]["items"]:
-            result[user["id"]] = {
+            # 'id' is likely available only in older JIRA Server, it's not available on newer JIRA Server.
```

Comment:
Honestly, looking at the existing docs I have no idea how long since they removed `id` . I looked as far back as v6.0.1 for Jira Server and it still didn't have an `id` property for the returned users.

![image](https://user-images.githubusercontent.com/26027314/123460283-4d864300-d5df-11eb-8289-8a83c6e5d339.png)


We can also see that the `group` endpoint is Deprecated, `group/member` is suggested instead (which will allow us to remove the legacy pagination in this function).

https://docs.atlassian.com/software/jira/docs/api/REST/8.13.6/#api/2/group-getUsersFromGroup
https://developer.atlassian.com/cloud/jira/platform/rest/v2/api-group-groups/#api-rest-api-2-group-get

Let's make a separate PR to deal with this, there we may remove `hasId` and instead have something like:

`result[user.get("accountId", user.get("name"))] =`